pil_base64_encode_block.py

```diff
@@ -0,0 +1,24 @@
+from base64 import b64encode
+from io import BytesIO
+
+from nio.block.base import Block
+from nio.properties import VersionProperty
+
+
+class PILBase64Encode(Block):
+
+    """ Convert a PIL image to a base64-encoded string """
+
+    version = VersionProperty('0.1.0')
+
+    def __init__(self):
+        super().__init__()
+        self.frame_buffer = BytesIO()
+
+    def process_signals(self, signals, input_id='default'):
+        for signal in signals:
+            signal.image.save(self.frame_buffer, format='JPEG')
+            signal.image = b64encode(self.frame_buffer.getvalue())
+            self.frame_buffer.truncate(0)
```

Comment:
Is it for some reason better to reuse the same `BytesIO` object and doing this every time? What's the tradeoff for creating a new `BytesIO` object instead of all the `truncate` `seek` stuff?

Reply:
Not sure, actually. I assumed that is was more preferred and/or better to reuse the same object -- I thought that creating the object would take longer than reusing it. What do you think? I guess there are tools for testing such things.

Reply:
for future reference, it turns out that creating new objects is just a tiny bit faster, and stream size has no real effect up to 1MB:
```>>> def new_instance(size=1):
...     foo = io.BytesIO()
...     foo.write(b'f' * size)
...
>>> def reuse_instance(size=1):
...     bar.seek(0)
...     bar.truncate()
...     bar.write(b'b' * size)
>>> bar = io.BytesIO()
>>> 
>>> timeit.timeit("new_instance()", setup="from __main__ import new_instance")
0.35561818895234865
>>> bar = io.BytesIO()
>>> timeit.timeit("reuse_instance()", setup="from __main__ import reuse_instance")
0.38163199666257697
>>> timeit.timeit("new_instance(100000)", setup="from __main__ import new_instance")
5.388827219830887
>>> timeit.timeit("reuse_instance(100000)", setup="from __main__ import reuse_instance")
5.398885070431561
>>> timeit.timeit("new_instance(1000000)", setup="from __main__ import new_instance")
60.53027565825846
>>> timeit.timeit("reuse_instance(1000000)", setup="from __main__ import reuse_instance")
62.16274857452095
>>>```